app/src/main/java/com/keylesspalace/tusky/components/login/LoginActivity.kt

```diff
@@ -228,7 +233,15 @@ class LoginActivity : BaseActivity(), Injectable {
             domain, clientId, clientSecret, oauthRedirectUri, code, "authorization_code"
         ).fold(
             { accessToken ->
-                accountManager.addAccount(accessToken.accessToken, domain)
+                if (intent.getIntExtra(LOGIN_MODE, MODE_DEFAULT) == MODE_MIGRATION) {
+                    accountManager.activeAccount?.let {
+                        it.accessToken = accessToken.accessToken
+                        it.oauthScopes = OAUTH_SCOPES
+                        accountManager.saveAccount(it)
```

Comment:
We cannot be sure the user logged in with the same account, they could have mulltiple accounts on the same instance so this is dangerous. Just using `accountManager.addAccount` should work though? `updateActiveAccount` will fix it if the account already exists